File: 24a0ff-HFT_bridge_bit-perfect_mechanical_refactor/evaluation/evaluation.py
#!/usr/bin/env python3
import sys
import json
import os
import uuid
import platform
import subprocess
from pathlib import Path
from datetime import datetime

ROOT = Path(__file__).resolve().parent.parent
REPORTS = ROOT / "evaluation" / "reports"

def environment_info():
    return {
        "python_version": platform.python_version(),
        "platform": platform.platform()
    }

def run_tests(target: str):
    env = os.environ.copy()
    env["TEST_TARGET"] = target
    try:
        # Using unified_test.py which handles loading the correct module via TEST_TARGET env var
        proc = subprocess.run(
            [sys.executable, str(ROOT / "tests" / "unified_test.py")],
            cwd=ROOT,
            capture_output=True,
            text=True,
            timeout=120,
            env=env
        )
        return {
            "passed": proc.returncode == 0,
            "return_code": proc.returncode,
            "output": (proc.stdout + proc.stderr)[:8000]
        }
    except subprocess.TimeoutExpired:
        return {
            "passed": False,
            "return_code": -1,
            "output": "timeout"
        }

def run_metrics(repo_path: Path):
    # Metric: Line count (crucial for this refactor task)
    file_path = repo_path / "hft_parity_refactor.py"
    if not file_path.exists():
        return {}
    
    with open(file_path, 'r') as f:
        line_count = len(f.readlines())
        
    return {
        "line_count": line_count
    }

def evaluate(repo_name: str):
    repo_path = ROOT / repo_name
    # Map repo name to test target (before/after)
    target = "before" if "before" in repo_name else "after"
    tests = run_tests(target)
    metrics = run_metrics(repo_path)
    return {
        "tests": tests,
        "metrics": metrics
    }

def run_evaluation():
    run_id = str(uuid.uuid4())
    start = datetime.utcnow()
    
    before = evaluate("repository_before")
    after = evaluate("repository_after")
    

    passed_gate = after["tests"]["passed"]
    
    improvement_summary = (
        f"Refactor successful. Line count: {before['metrics'].get('line_count')} -> {after['metrics'].get('line_count')}. "
        "All behavioral and structural tests passed in the 'after' repository."
    )
    
    end = datetime.utcnow()
    return {
        "run_id": run_id,
        "started_at": start.isoformat() + "Z",
        "finished_at": end.isoformat() + "Z",
        "duration_seconds": (end - start).total_seconds(),
        "environment": environment_info(),
        "before": before,
        "after": after,
        "comparison": {
            "passed_gate": passed_gate,
            "improvement_summary": improvement_summary
        },
        "success": passed_gate,
        "error": None
    }

def main():
    REPORTS.mkdir(parents=True, exist_ok=True)
    report = run_evaluation()
    
    # Write latest.json
    path = REPORTS / "latest.json"
    path.write_text(json.dumps(report, indent=2))
    
    print(f"Report written to {path}")
    return 0 if report["success"] else 1

if __name__ == "__main__":
    sys.exit(main())

File: 6b24e6-JSON_schema_validator/evaluation/evaluation.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');
const { v4: uuidv4 } = require('crypto').randomUUID ? { v4: () => require('crypto').randomUUID() } : require('uuid');

const ROOT = path.join(__dirname, '..');
const REPORTS = path.join(__dirname, 'reports');

function getUUID() {
    try {
        return require('crypto').randomUUID();
    } catch {
        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
            const r = Math.random() * 16 | 0;
            const v = c === 'x' ? r : (r & 0x3 | 0x8);
            return v.toString(16);
        });
    }
}

function environmentInfo() {
    const nodeVersion = process.version;
    const platform = `${process.platform}-${process.arch}`;
    return {
        node_version: nodeVersion,
        platform: platform
    };
}

function runTests(repoName) {
    const testArg = repoName === 'repository_before' ? 'before' : 'after';
    
    try {
        const result = execSync(
            `node tests/test_all.js ${testArg}`,
            {
                cwd: ROOT,
                encoding: 'utf8',
                timeout: 120000,
                stdio: 'pipe'
            }
        );
        
        return parseTestOutput(result, 0);
    } catch (error) {
        if (error.stdout || error.stderr) {
            const output = (error.stdout || '') + (error.stderr || '');
            return parseTestOutput(output, error.status || 1);
        }
        return {
            passed: false,
            return_code: -1,
            output: `Test execution error: ${error.message}`,
            test_count: 0,
            passed_count: 0,
            failed_count: 0
        };
    }
}

function parseTestOutput(output, returnCode) {
    let passedCount = 0;
    let failedCount = 0;
    let totalCount = 0;
    
    const lines = output.split('\n');
    for (const line of lines) {
        if (line.includes('Passed:')) {
            const match = line.match(/Passed:\s*(\d+)/);
            if (match) passedCount = parseInt(match[1], 10);
        }
        if (line.includes('Failed:')) {
            const match = line.match(/Failed:\s*(\d+)/);
            if (match) failedCount = parseInt(match[1], 10);
        }
        if (line.includes('Total:')) {
            const match = line.match(/Total:\s*(\d+)/);
            if (match) totalCount = parseInt(match[1], 10);
        }
    }
    
    return {
        passed: returnCode === 0,
        return_code: returnCode,
        output: output.slice(0, 8000),
        test_count: totalCount,
        passed_count: passedCount,
        failed_count: failedCount
    };
}

function runMetrics(tests) {
    const passRate = tests.test_count > 0 
        ? Math.round((tests.passed_count / tests.test_count) * 10000) / 100
        : 0.0;
    
    return {
        pass_rate: passRate,
        test_count: tests.test_count,
        passed_count: tests.passed_count,
        failed_count: tests.failed_count
    };
}

function evaluate(repoName) {
    const tests = runTests(repoName);
    const metrics = runMetrics(tests);
    return {
        tests: tests,
        metrics: metrics
    };
}

function runEvaluation() {
    const runId = getUUID();
    const start = new Date();
    
    try {
        const before = evaluate('repository_before');
        const after = evaluate('repository_after');
        
        const improvement = after.metrics.passed_count - before.metrics.passed_count;
        
        let summary;
        if (improvement > 0) {
            summary = `Fixed ${improvement} bug(s). After: ${after.metrics.passed_count}/${after.metrics.test_count} tests pass.`;
        } else if (improvement < 0) {
            summary = `Regression: ${Math.abs(improvement)} test(s) now failing.`;
        } else {
            summary = `No change. ${after.metrics.passed_count}/${after.metrics.test_count} tests pass.`;
        }
        
        const comparison = {
            passed_gate: after.tests.passed,
            improvement_summary: summary
        };
        
        const end = new Date();
        
        return {
            run_id: runId,
            started_at: start.toISOString(),
            finished_at: end.toISOString(),
            duration_seconds: (end - start) / 1000,
            environment: environmentInfo(),
            before: before,
            after: after,
            comparison: comparison,
            success: comparison.passed_gate,
            error: null
        };
    } catch (error) {
        const end = new Date();
        return {
            run_id: runId,
            started_at: start.toISOString(),
            finished_at: end.toISOString(),
            duration_seconds: (end - start) / 1000,
            environment: environmentInfo(),
            before: null,
            after: null,
            comparison: null,
            success: false,
            error: error.message
        };
    }
}

function main() {
    if (!fs.existsSync(REPORTS)) {
        fs.mkdirSync(REPORTS, { recursive: true });
    }
    
    const report = runEvaluation();
    
    // Generate timestamp filename: YYYY-MM-DD_HH-MM-SS.json
    const now = new Date();
    const timestamp = now.toISOString()
        .replace(/:/g, '-')
        .replace(/\..+/, '')
        .replace('T', '_');
    const timestampFile = `${timestamp}.json`;
    
    // Write timestamped report
    const reportPath = path.join(REPORTS, timestampFile);
    fs.writeFileSync(reportPath, JSON.stringify(report, null, 2));
    
    // Also write latest.json for convenience
    const latestPath = path.join(REPORTS, 'latest.json');
    fs.writeFileSync(latestPath, JSON.stringify(report, null, 2));
    
    console.log(`Report written to ${reportPath}`);
    console.log(`Latest copy: ${latestPath}`);
    console.log('\nEvaluation Summary:');
    console.log(`  Run ID: ${report.run_id}`);
    console.log(`  Success: ${report.success}`);
    
    if (report.comparison) {
        console.log(`  ${report.comparison.improvement_summary}`);
    }
    
    if (report.error) {
        console.log(`  Error: ${report.error}`);
    }
    
    return report.success ? 0 : 1;
}

if (require.main === module) {
    process.exit(main());
}

module.exports = { runEvaluation, main };

